arcade/examples/tetris.py

```diff
@@ -176,105 +251,129 @@ def drop(self):
+        if not self.game_over:      # and not self.paused:
+            new_stone = rotate_counter_clockwise(self.stone)
+            if not self.board.check_collision(
+                    new_stone, self.stone_x, self.stone_y):
                 self.stone = new_stone
+        return
 
     def on_update(self, dt):
-        """ Update, drop stone if warrented """
-        self.frame_count += 1
-        if self.frame_count % 10 == 0:
-            self.drop()
-
-    def move(self, delta_x):
+        """ Update, drop stone if warranted """
+        now = time.time()
+        if now >= self._time_next_update:
+            self._time_next_update = now + self._desired_update_delay
+            self.drop_stone()
+        # super().on_update(dt)
+        return
+
+    def move_stone(self, delta_x):
         """ Move the stone back and forth based on delta x. """
-        if not self.game_over and not self.paused:
+        if not self.game_over:  # and not self.paused:
             new_x = self.stone_x + delta_x
             if new_x < 0:
                 new_x = 0
-            if new_x > COLUMN_COUNT - len(self.stone[0]):
-                new_x = COLUMN_COUNT - len(self.stone[0])
-            if not check_collision(self.board, self.stone, (new_x, self.stone_y)):
+            elif new_x > self.board._columns - len(self.stone[0]):
+                new_x = self.board._columns - len(self.stone[0])
+            if not self.board.check_collision(self.stone, new_x, self.stone_y):
                 self.stone_x = new_x
+        return
 
     def on_key_press(self, key, modifiers):
         """
         Handle user key presses
+        as in https://strategywiki.org/wiki/Tetris/Controls#General_controls
         User goes left, move -1
         User goes right, move 1
         Rotate stone,
         or drop down
         """
-        if key == arcade.key.LEFT:
-            self.move(-1)
-        elif key == arcade.key.RIGHT:
-            self.move(1)
-        elif key == arcade.key.UP:
-            self.rotate_stone()
-        elif key == arcade.key.DOWN:
-            self.drop()
+        if key in (arcade.key.LEFT, arcade.key.NUM_LEFT):
+            self.move_stone(-1)
+        elif key in (arcade.key.RIGHT, arcade.key.NUM_RIGHT):
+            self.move_stone(1)
+        elif key in (arcade.key.UP, arcade.key.X, arcade.key.NUM_UP):
+            self.rotate_stone_cw()
+        elif key in (arcade.key.LCTRL, arcade.key.RCTRL, arcade.key.Z):
+            self.rotate_stone_ccw()
+        elif key in (arcade.key.DOWN, arcade.key.NUM_DOWN):
+            self.drop_stone()
+        elif key == arcade.key.SPACE:
+            self.drop_stone_hard()
+        elif key == arcade.key.P:
+            self.paused = not self.paused
+        elif key in (arcade.key.PLUS, arcade.key.NUM_ADD):
+            self.change_update_rate(1)
+        elif key in (arcade.key.MINUS, arcade.key.NUM_SUBTRACT):
+            self.change_update_rate(-1)
+        return
```

Comment:
This adds a lot of options. I'd suggest pushing to a larger stand-alone or a step-by-step tutorial.